types/csmim.obj.directory.1.yaml

```diff
@@ -0,0 +1,57 @@
+%YAML 1.2
+---
+id: csmim.obj.directory.1
+supertypes: []
+description: >-
+  CSMIM resource directory service. The resource directory announces
+  registration and removal of objects in the CSMIM network depending on the
+  types of the objects. To this avail, the resource directory has an internal
+  list of "object types of interest". Whenever an object is registered which is
+  derived from one of these types of interest, the resource directory publishes
+  a message to announce the path of that object. It does so using the MQTT
+  extended topic feature of ARINC 853, publishing the message under the
+  "csmim.obj" resource. When an object vanishes which was derived from one of
+  the types of interest, the resource directory publishes an empty message to
+  the same extended topic.
+
+  There is currently no mechanism for a client to add an object type to the
+  list of types of interest. The list is statically configured. It should
+  contain only few entries so as to reduce the number of messages that have to
+  be published by the resource directory, in particular at network startup.
+
+  The resource directory always resides at the path "core/csmim/directory".
```

Comment:
why not `v1/core/directory/` as `csmim.obj` is already following in the topic?

Reply:
I wrote the *path* there. Should I instead give the MQTT topic name "v1/data/core/csmim/directory"? Same question for the example below.

Reply:
ah, you are right. I have implemented registration at `core/registration` but it is defined as `core/csmim/registration`.